client/src/app/auth/auth.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { AuthService } from './services/auth.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+
+  constructor(private authservice: AuthService) {}
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    let token = localStorage.getItem('token')
```

Comment:
pls encapsulate this logic in auth service. create a method getToken() there.
auth interceptor shouldn't know about how and where you store the token

Reply:
I created getToken() function inside of auth service